src/app/clusters/color-control-server/color-control-server.cpp

```diff
@@ -25,11 +25,303 @@
 #include <platform/CHIPDeviceLayer.h>
 #include <platform/PlatformManager.h>
 
+#ifdef EMBER_AF_PLUGIN_SCENES
+#include <app/clusters/scenes-server/scenes-server.h>
+#endif
+
 using namespace chip;
 using namespace chip::app::Clusters;
 using namespace chip::app::Clusters::ColorControl;
 using chip::Protocols::InteractionModel::Status;
 
+#ifdef EMBER_AF_PLUGIN_SCENES
+class DefaultColorControlSceneHandler : public scenes::DefaultSceneHandlerImpl
+{
+public:
+    enum ColorControlEFS : uint8_t
+    {
+        kCurrentX = 0,
+        kCurrentY,
+        kEnhancedCurrentHue,
+        kCurrentSaturation,
+        kColorLoopActive,
+        kColorLoopDirection,
+        kColorLoopTime,
+        kColorTemperatureMireds,
+        kEnhancedColorMode,
+    };
+
+    // As per spec, 2 attributes are scenable in the level control cluster
+    static constexpr uint8_t kColorControlMaxScenableAttributes = 9;
+
+    DefaultColorControlSceneHandler() = default;
+    ~DefaultColorControlSceneHandler() override {}
+
+    // Default function for ColorControl cluster, only puts the ColorControl cluster ID in the span if supported on the caller
+    // endpoint
+    void GetSupportedClusters(EndpointId endpoint, Span<ClusterId> & clusterBuffer) override
+    {
+        ClusterId * buffer = clusterBuffer.data();
+        if (emberAfContainsServer(endpoint, ColorControl::Id) && clusterBuffer.size() >= 1)
```

Comment:
If this tests false, shouldn't we reduce the size of the buffer to 0?  Otherwise we are returning whatever size it has incoming, with garbage inside....